### DISCUSSION TITLE
Want make more clear why get rid of ESM?

### DISCUSSION THREAD

**Discussion by HillLiu** [testing-library/user-event#1021] [Q&A] [answered]
Check these issues.
https://github.com/testing-library/user-event/pull/949/files
https://github.com/testing-library/user-event/issues/839

I saw the following code was removed.
```
  "main": "./dist/index.cjs",
  "module": "./dist/index.mjs",
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "default": "./dist/index.mjs"
    }
  },
```

When remove these codes (exports xxx), in ESM it will think it is a CJS module. 

So there are two different ways.

* pure ESM module.
```
import userEvent from "@testing-library/user-event";
console.log(userEvent.setup);
```

* use CJS in ESM
```
import userEvent from "@testing-library/user-event";
console.log(userEvent.default.setup);
```

Actually I could treat @testing-library/user-event as a CJS library,  and I have a smart util function to check if xxx.default exists or not.

However as I know the Native ESM should be the future, I want make sure we intent to do it (Go cjs), and it is not an unexpected change.

---

BTW for issue 839, I guess the solution should use correct file extensions for type script such as .mts or .cts.
check it. https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#new-file-extensions

or add type: "module" in package.json.
check it. https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#type-in-package-json-and-new-extensions

**Comment by ph-fritsche** [comment, marked as answer]
Fixed in #1022 - Thanks for the report :heart:

**Reply by ph-fritsche** [reply]
@all-contributors add @HillLiu bug
